proof/refine/RISCV64/Invariants_H.thy

```diff
@@ -751,9 +1054,14 @@ definition
   valid_queues_no_bitmap :: "kernel_state \<Rightarrow> bool"
 where
  "valid_queues_no_bitmap \<equiv> \<lambda>s.
-   (\<forall>d p. (\<forall>t \<in> set (ksReadyQueues s (d, p)). obj_at' (inQ d p and runnable' \<circ> tcbState) t s)
-    \<and>  distinct (ksReadyQueues s (d, p))
-    \<and> (d > maxDomain \<or> p > maxPriority \<longrightarrow> ksReadyQueues s (d,p) = []))"
+   (\<forall>d p. (\<forall>t \<in> set (ksReadyQueues s (d, p)). obj_at' (inQ d p) t s)
+          \<and> (d > maxDomain \<or> p > maxPriority \<longrightarrow> ksReadyQueues s (d,p) = []))"
+
+defs ready_qs_runnable_def:
+  "ready_qs_runnable \<equiv> \<lambda>s. \<forall>d p. \<forall>t \<in> set (ksReadyQueues s (d, p)). st_tcb_at' runnable' t s"
+
+definition notQueued :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
+  "notQueued t s \<equiv> (\<forall>a b. t \<notin> set (ksReadyQueues s (a, b)))"
```

Comment:
remove outer parentheses

Reply:
In fact, this should not have `a b ` and then pass `(a,b)` to ksReadyQueues. If we really want two components here, it should be `d` and `p`, and I would usually recommend just a single variable for the entire pair. Let's not do the latter in this case, because the proofs are going to assume two now and if we fix this here, we're going to have to fix it in the proofs, which we should only do once we have them working.